packages/core/phase2/lib/getOperationSequence/areAllResultsAlreadyPresentOnPnc/isMatchToPncAdjAndDis/isMatchToPncAdjAndDis.ts

```diff
@@ -1,57 +1,52 @@
-import type { AnnotatedHearingOutcome, Result } from "../../../../../types/AnnotatedHearingOutcome"
+import type { AnnotatedHearingOutcome, Offence } from "../../../../../types/AnnotatedHearingOutcome"
 import type Exception from "../../../../../types/Exception"
 import type { ExceptionResult } from "../../../../../types/Exception"
-import type { NonEmptyArray } from "../../../../../types/NonEmptyArray"
-import allRecordableResultsMatchAPncDisposal from "./allRecordableResultsMatchAPncDisposal"
-import getAdjFromAho from "./getAdjFromAho"
-import isMatchToPncAdj from "./isMatchToPncAdj"
+import { isNonEmptyArray } from "../../../../../types/NonEmptyArray"
+import areResultsMatchAPncDisposal from "./areResultsMatchAPncDisposal"
+import createPncAdjudicationByAho from "./createPncAdjudicationByAho"
+import isMatchToPncAdjudication from "./isMatchToPncAdjudication"
 
-const isMatchToPncAdjAndDis = (
-  results: NonEmptyArray<Result>,
-  aho: AnnotatedHearingOutcome,
-  courtCaseReferenceNumber: string | undefined,
-  offenceIndex: number,
-  offenceReasonSequence: string | undefined
-): ExceptionResult<boolean> => {
+const findPncCourtCase = (aho: AnnotatedHearingOutcome, offence: Offence) => {
   const courtCaseReference =
-    courtCaseReferenceNumber ?? aho.AnnotatedHearingOutcome.HearingOutcome.Case.CourtCaseReferenceNumber
+    offence.CourtCaseReferenceNumber ?? aho.AnnotatedHearingOutcome.HearingOutcome.Case.CourtCaseReferenceNumber
 
-  if (!courtCaseReference) {
-    return { value: false, exceptions: [] }
-  }
-
-  const courtCase = aho.PncQuery?.courtCases?.find((x) => x.courtCaseReference === courtCaseReference)
+  return courtCaseReference
+    ? aho.PncQuery?.courtCases?.find((x) => x.courtCaseReference === courtCaseReference)
+    : undefined
+}
 
-  if (!courtCase) {
+const isMatchToPncAdjAndDis = (
+  aho: AnnotatedHearingOutcome,
+  offence: Offence,
+  offenceIndex: number
+): ExceptionResult<boolean> => {
+  if (!offence.Result || !isNonEmptyArray(offence.Result)) {
```

Comment:
I feel like if we have a `isNonEmptyArrray`, we should have a `isEmptyArray` to avoid have to do `!isNonEmptyArrray` so it reads nicer.

Reply:
I don't think we can have the `isEmptyArray()` function.
`isNonEmptyArray()` uses type guard to indicate whether the type is `NonEmptyArray<T>`. This is needed because we use this type in one of the nested functions.
```ts
const isNonEmptyArray = <T>(arr: T[]): arr is NonEmptyArray<T> => {
  return arr.length > 0
}
```
We might be able to refactor and remove this type though, as it's been used in one function only.

Reply:
Ahhhhh I see, no worries then!